### PR TITLE
Enable openshift tests

### PR DESCRIPTION
**Pull request by rmohr**
Start running functional tests on openshift.

Signed-off-by: Roman Mohr <rmohr@redhat.com>

### CODE REVIEW

**Comment by rmohr** [comment]
ok to test

**Comment by rmohr** [comment]
retest this please

**Comment by rmohr** [comment]
retest this please

File: cluster/deploy.sh [hunk 1]
```diff
@@ -26,9 +26,9 @@ source hack/config.sh
 echo "Deploying ..."
 
 # Deploy the right manifests for the right target
-if [ -z "$TARGET" ] || [ "$TARGET" = "vagrant-dev" ]; then
+if [ -z "$TARGET" ] || [ "$TARGET" = "vagrant-dev" ] || [ "$TARGET" = "kubernetes-dev" ] ; then
```
Comment (davidvossel) [inline cluster/deploy.sh]:
```
if [ -z "$TARGET ] ||  [[ $TARGET =~ .*-dev ]]; then
```

File: cluster/deploy.sh [hunk 2]
```diff
@@ -26,9 +26,9 @@ source hack/config.sh
...
     _kubectl create -f ${MANIFESTS_OUT_DIR}/dev -R $i
-elif [ "$TARGET" = "vagrant-release" ]; then
+elif [ "$TARGET" = "vagrant-release" ] || [ "$TARGET" = "openshift-release" ] || [ "$TARGET" = "kubernetes-release" ] ; then
```
Comment (davidvossel) [inline cluster/deploy.sh]: if [[ $TARGET =~ .*-release ]]; then

File: automation/test.sh [hunk 3]
```diff
@@ -19,14 +19,23 @@
...
+#     kubernetes-release: Runs all functional tests on a release kubernetes setup
+#     openshift-release: Runs all functional tests on a release openshift setup
 #     TODO: vagrant-tagged-release: Runs all possible functional tests on a release deployment in vagrant on a tagged release
 
 set -ex
 
 export WORKSPACE="${WORKSPACE:-$PWD}"
-export PROVIDER="k8s-1.9.3"
+
+if [ "$TARGET" = "openshift-release" ]; then
```
Comment (davidvossel) [inline automation/test.sh]:
```
if [[ $TARGET =~ openshift-.* ]]; then
```

**Comment by rmohr** [comment]
retest this please

**Comment by cynepco3hahue** [comment]
The code now looks fine, it succeeds to deploy all necessary components and starts tests, so from that perspective we can merge it.
But it has problems to run tests on the multi-node environment
```
./cluster/kubectl.sh console -n kubevirt-test-default testvmq75fv
Escape sequence is ^]
                     error dialing backend: dial tcp: lookup node02 on 192.168.66.101:53: no such host
```
it just uses OpenShift DSN that does not aware of the host with name node02, so until I will fix it I will leave this job to disable under Jenkins.

**Comment by rmohr** [comment]
> The code now looks fine, it succeeds to deploy all necessary components and starts tests, so from that perspective we can merge it.
> But it has problems to run tests on the multi-node environment

Ok the problem is the same like we had in general with dns (remember the extra dhclient in the script?). Once dhclient was run again and i do `systemctl restart origin-master-api.service` it works like expected. We can try to fix the underlying problem that dns does not seem to be run anymore once openshift is installed.

**Comment by cynepco3hahue** [comment]
@rmohr I already fixed by adding explicitly host records under `dnsmasq` config that OpenShuft uses, 
give me know if it looks fine for you and I will merge the PR - https://github.com/rmohr/qemu-dockerized/pull/27

**Comment by cynepco3hahue** [comment]
@rmohr @davidvossel OpenShift build looks better after applying dnsmasq fix - https://rfenkhuber.fedorapeople.org/jenkins/3/openshift-test-console.log

**Comment by rmohr** [comment]
lgtm. Can't approve a PR I opened in github.

**Pull request merged**